src/server/server/dal.py

```diff
@@ -14,10 +14,17 @@ def get_user(email):
     db_user = db.User.get(email)
     return _to_user(db_user) if db_user else None
 
+def get_user_by_checking_password(email, password):
+    user = get_user(email)
+    if user is not None and user.password == password:
+        return user
+    return None
 
```

Comment:
I don't think the check of the password belongs to the DAL, I would put it in the server side.

If you change the database, you change the dal but you would do the exact same password check